resource-watch/resource-watch
# FEAT / custom layer interaction

## Overview
Allow admin users to add custom interactions, show warning so they know its a custom one when selecting format.

* Fix data type issues breaking the map tooltip
* When updating a layer, you now stay on the same page 
 * When entering layer admin, the tooltip now works when no interactions are present on init

Updated UI 

New button under layer config to save the form, also now we verify that the layer config is valid before submit.  (see error example in screenshot)
![screen shot 2018-04-05 at 15 58 39](https://user-images.githubusercontent.com/971129/38372615-8e2e99b0-38ef-11e8-921d-049c39c83d72.png)

If you click "verify" layer config and its valid, you will get success feedback
![screen shot 2018-04-06 at 12 07 54](https://user-images.githubusercontent.com/971129/38415898-2e0f4a08-3993-11e8-8c9a-5e1071ed0e32.png)

Custom interactions now have a different UI as they can easily break the query. 
![screen shot 2018-04-05 at 16 39 56](https://user-images.githubusercontent.com/971129/38372823-0362df48-38f0-11e8-8a9c-0055aecd6a02.png)


If you add a custom interaction, and try to save, while that field is NOT available in the **query** you wont be available to save the layer.

## Testing instructions
`/admin/data/layers` start typing in the field and if we dont have the interactions you should be prompted to add it. 

All formats should be available in the format select input for the custom ones, also when focusing you should see a warning that its a custom one without data type. 

## Pivotal task
https://www.pivotaltracker.com/story/show/156460988
